kratos/utilities/math_utils.h

```diff
@@ -1876,6 +1876,56 @@ class MathUtils
         return is_converged;
     }
 
+    /**
+     * @brief Calculates the Factorial of a number k, Factorial = k!
+     * @tparam Number The number of which the Factorial is computed
+     */
+    template<class TIntegerType>
+    static inline TIntegerType Factorial(
+        const TIntegerType Number
+        )
+    {
+        if (Number == 0) {
+            return 1;
+        }
+        TIntegerType k = Number;
+        for (unsigned int i = Number - 1; i > 0; --i){
+            k *= i;
+        }
+        return k;
+    }
+
+    /**
+     * @brief Calculates the exponential of a matrix
+     * @brief see https://mathworld.wolfram.com/MatrixExponential.html
+     * @tparam rMatrix: the matrix A of which exp is calculated
+     * @tparam rExponentialMatrix: exp(A)
+     */
+    template<class TMatrixType>
+    static inline void CalculateExponentialOfMatrix(
+            const TMatrixType& rMatrix,
+            TMatrixType& rExponentialMatrix
+        )
+    {
+        int series_term = 2, max_terms = 200, factorial = 1;
```

Comment:
```suggestion
        std::size_t series_term = 2;
        std::size_t factorial = 1;
        const std::size_t max_terms = 200;
```
Shouldn't we expose the `max_terms` to the user (defaulting it to the current value)?